### PR TITLE
CPDTP-205 Ensure no payment is made where a declaration has been voided

### PR DESCRIPTION
**Pull request by plisovin**
## Ticket and context

Ticket: https://dfedigital.atlassian.net/browse/CPDTP-205

## Tech review

As the voided declarations are already not included in the query scope used in the aggregation then only a feature test is required to ensure the payment breakdown doesn't include a voided declaration 

### Code quality checks
- [ ] All commit messages are meaningful and true
- [ ] Added enough automated tests

### HTML Checks
- [ ] All new pages have automated accessibility checks
- [ ] All new pages have visual tests via Percy

### Gotchas
- [ ] All `School` queries are correctly scoped - `eligible` when they need to be

### How can someone see it it review app?
1. Click the link to review app (posted by a `github-actions` bot below)
2. Follow the steps from the ticket.

## External API changes

Does this change anything in our external APIs? If so, did you remember to update the CHANGELOG for Lead Providers? (lead-providers/guidance/release-notes)

### CODE REVIEW

File: spec/features/participant-declarations/voided_declaration_in_payment_breakdown_spec.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  before(:each) do
+    setup
+  end
+
+  scenario "Payment breakdown does not include voided declarations" do
+    given_an_early_career_teacher_has_been_entered_onto_the_dfe_service
+    and_call_off_contract_was_created_for_lead_provider
+    when_the_participant_details_are_passed_to_the_lead_provider
+    @declaration_id = and_the_lead_provider_submits_a_declaration_for_the_ect_using_their_id.dig("data", "id")
```
Comment (kicferk1) [inline spec/features/participant-declarations/voided_declaration_in_payment_breakdown_spec.rb:17]: Can we make `@ declaration_id` part of `and_the_lead_provider_submits_a_declaration_for_the_ect_using_their_id` step?

File: spec/features/participant-declarations/voided_declaration_in_payment_breakdown_spec.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    )
+  end
+
+  def and_profile_declaration_made_payable
+    profile_declaration = ProfileDeclaration.find_by(participant_declaration_id: @declaration_id)
+    profile_declaration.update!({ payable: true })
+  end
+
+  def then_the_payment_breakdown_does_not_include_voided_declaration
+    expect(@breakdown.dig(:breakdown_summary, :ects)).to eq(0)
```
Comment (kicferk1) [inline spec/features/participant-declarations/voided_declaration_in_payment_breakdown_spec.rb:44]: Maybe we want `participants` here - the user we make a declaration against could be a mentor.

File: spec/features/participant-declarations/voided_declaration_in_payment_breakdown_spec.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    @breakdown = CalculationOrchestrator.call(
+      cpd_lead_provider: @cpd_lead_provider,
+      contract: @cpd_lead_provider.lead_provider.call_off_contract,
+      event_type: :started,
+    )
+  end
+
+  def and_profile_declaration_made_payable
+    profile_declaration = ProfileDeclaration.find_by(participant_declaration_id: @declaration_id)
+    profile_declaration.update!({ payable: true })
```
Comment (kicferk1) [inline spec/features/participant-declarations/voided_declaration_in_payment_breakdown_spec.rb:40]: No need to pass a hash explicitly, right? `profile_declaration.update!(payable: true)` should do fine.

File: app/serializers/participant_declaration_serializer.rb [hunk 4]
```diff
@@ -10,7 +10,7 @@ class ParticipantDeclarationSerializer
   set_type :'participant-declaration'
   attributes :participant_id, :declaration_type, :course_identifier, :voided
 
-  attribute :eligible_for_payment do |declaration|
+  attribute :participant_eligible_for_funding do |declaration|
```
Comment (kicferk1) [inline app/serializers/participant_declaration_serializer.rb:13]: That does not sound like it wants to be in this PR, right?

Comment (plisovin) [reply app/serializers/participant_declaration_serializer.rb:13]: oops, was looking at something else

Comment (plisovin) [reply app/serializers/participant_declaration_serializer.rb:13]: will roll back this commit

**Review by kicferk1 — APPROVED** [review]

**Pull request merged**